### DISCUSSION TITLE
How to keep results always fresh (prevent them from going stale)?

### DISCUSSION THREAD

**Discussion by alexbepple** [TanStack/query#230] [General] [answered]
The only way I have found is to set `staleTime` to a very high value. Surprisingly, though, `Number.MAX_SAFE_INTEGER` did not work. react-query kept revalidating.

Setting `staleTime` to a very high value bothers me slightly because I need to concern myself with incidental details.

Is there a clean way to effectively set `staleTime` to infinite?

**Comment by tannerlinsley** [comment]
Just set it to a large amount of time that would surpass the life of your users session length.

`staleTime: 1000 * 60 * 60 * 24 * 365 // One year`

**Comment by tannerlinsley** [comment]
Also, this should have been posted in the Discussions tab. :) I'll convert it to that now.

**Comment by alexbepple** [comment]
Sorry about "issues" vs "discussions".

In your example, `staleTime` is an incidental detail. I think it would be preferable to have an explicit way of saying "forever".

**Comment by tannerlinsley** [comment]
While I see where you're coming from, however at the end of the day, it would eventually need to be transformed into a time duration since the whole thing runs on timers for cache management. This would be quite a bit of library overhead to write where a simple constant in your ReactQueryConfigProvider or custom hook that wraps useQuery would be just fine.

**Comment by cherniavskii** [comment]
Actually we can support `Infinity` value for `staleTime`, and not set timeout at all in this case.
What do you think?

**Comment by tannerlinsley** [comment]
As long as it doesn't create a bunch of complexity or unnecessary KBs, I'm fine with that.

**Comment by cherniavskii** [comment, marked as answer]
Here's a PR https://github.com/tannerlinsley/react-query/pull/232

**Comment by alexbepple** [comment]
😍 Thank you, @cherniavskii and @tannerlinsley.
